apps/src/music/views/MiniMusicPlayer.tsx

```diff
@@ -35,15 +37,22 @@ const MiniPlayerView: React.FunctionComponent<MiniPlayerViewProps> = ({
   const [currentProjectId, setCurrentProjectId] = useState<string | undefined>(
     undefined
   );
+  const {userId, userType, signInState} = useAppSelector(
+    state => state.currentUser
+  );
+  const analyticsReporter = useMemo(() => new AnalyticsReporter(), []);
 
-  // Setup library and workspace on mount
+  // Setup library, workspace, and analyticsReporter on mount
   const onMount = useCallback(async () => {
     setUpBlocklyForMusicLab();
     workspaceRef.current.initHeadless();
     const library = await loadLibrary(libraryName);
     MusicLibrary.setCurrent(library);
+    analyticsReporter.startSession().then(() => {
+      analyticsReporter.setUserProperties(userId, userType, signInState);
+    });
     setIsLoading(false);
```

Comment:
this is probably fine, but just noting that these'll be done out of sequence - it'll fire `setIsLoading(false)`, and _then_ do the results of the promise succeeding and fire `analyticsReporter.setUserProperties(userId, userType, signInState);`

I don't know if it matters for anything going on here.

If it _does_ matter, you can just move the `setIsLoading(false)` call to the end of the `then` of the `startSession()` promise so it'll fire after `analyticsReporter.setUserProperties`.

Reply:
Looks like `isLoading` is used to set loading UI while fetching the library so I'll go ahead and just move it after `MusicLibrary.setCurrent(library);`. Thanks!